happi/tests/test_cli.py

```diff
@@ -139,12 +143,16 @@ def assert_match_expected(
 
 def assert_in_expected(
     result: click.testing.Result,
-    expected_inclusion: str
+    expected_inclusion: str,
+    expected_return: int = 0
 ):
-    """standard checks for a cli result, confirms output includes expected"""
+    """
+    standard checks for a cli result, confirms output matches expected
+    SystemExit exception is acceptable if the expected return code is some failure code
+    """
     logger.debug(result.output)
-    assert result.exit_code == 0
-    assert not result.exception
+    assert result.exit_code == expected_return
+    assert not result.exception or (type(result.exception) is SystemExit and expected_return != 0)
```

Comment:
`type() is xyz` checks are generally avoided - use `isinstance` instead:
```suggestion
    assert not result.exception or (isinstance(result.exception, SystemExit) and expected_return != 0)
```